bin/common.ml

```diff
@@ -1235,3 +1235,9 @@ let context_arg ~doc =
     value
     & opt Arg.context_name Dune_engine.Context_name.default
     & info [ "context" ] ~docv:"CONTEXT" ~doc)
+
+let context_arg_opt ~doc =
```

Comment:
I'm not sure there is any point putting this in common since it is quite specialised. I would probably just inline it where you use it.

Reply:
I was following suit since the non-optional version of the argument is defined above.

Reply:
Right, but the non-optional version is used in multiple places, hence it is common. From what I can see, the only reason you want this "optional" version is so that you can check mutual exclusivity with the other flag. This is really a deficiency with cmdliner so it is understandable you had to go this way. However, I don't imagine any other tools using this argument hence my original suggestion.